src/parser.rs

```diff
@@ -295,9 +348,13 @@ where
         if self.peek() == Some(TOKEN_AT) {
             let kind = if bound { NODE_ERROR } else { NODE_PAT_BIND };
             self.start_node(kind);
+            let start = self.get_text_position();
             self.bump();
             self.expect_ident();
-            self.finish_node();
+            let end = self.finish_error_node();
+            if kind == NODE_ERROR {
```

Comment:
We probably want to use `if bound` here :)

Reply:
Will change.

Reply:
Fixed.